[Kb-Tay] iP

Kb-Tay opened the pull request:
### Obi-wan
A friendly Chat Bot that manages your tasks. 
> Hello There! - Obi-wan

Why use Obi-Wan? 
- Optimised for CLI
- Easy to use 
- ~~Simple~~ **SUPER** simple to use 

Features: 
- [x] Manage Task
- [ ] Find task by keyword (coming soon)
- [ ] GUI (in progress)

Instructions to run:
1. [Download Current Version](https://github.com/Kb-Tay/ip/releases/tag/v0.1)
2. `cd` to directory containing the `.jar` file.
3. Run `java -jar duke.jar` 


--- 
You can also learn OOP principles by reviewing the code :trollface:. This is a preview of the main method. 

```java
public static void main(String[] args) {
        new Duke("./src/main/data/duke.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Kb-Tay committed:
Renaming chatbot

Kb-Tay committed:
Add echo function

Kb-Tay committed:
Add tasklist

Kb-Tay committed:
Add Task Class, mark tasks

Kb-Tay committed:
Added tasks subclasses

Kb-Tay committed:
Add Testing

Kb-Tay committed:
Error handling

Kb-Tay committed:
Add delete function

Kb-Tay committed:
Add javadocs. Update error handling

Kb-Tay committed:
Add functionality to read and write tasks into txt file.

The program does not currently save the user input after program is stopped.

To save user input, add functions to read previously stored task list from txt file. Also add function to write new task list into the txt file.

Kb-Tay committed:
Merge branch 'branch-Level-7'

Kb-Tay committed:
Update chatbot to recognise DateTime inputs.

The addTask function accepted any string as a DateTime field previously.

Add validation and formatting of DateTime inputs.

Kb-Tay committed:
Clean up addTask function. Add javadocs for functions

Kb-Tay committed:
Merge branch 'branch-Level-8'

Kb-Tay committed:
Update code to follow OOP principles.

Code was previously cluster in the main class and could make improvements in encapsulation.

Multiple classes created to encapsulate user interactions, reading and writing files, understanding user input and the task list.

This cleans up and improves code quality.

Kb-Tay committed:
Update javadocs and clean up code

Kb-Tay committed:
Merge branch 'A-MoreOOP'

Kb-Tay committed:
Update classes into separate packages

Kb-Tay committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Kb-Tay committed:
Add Junit testing for Parse and TaskList class.

Kb-Tay committed:
Merge branch 'branch-A-Junit'

Kb-Tay committed:
Update error message and build.gradle file.

Kb-Tay committed:
Update javadocs.

Kb-Tay committed:
Add find command to search tasks.

User unable to filter task based on description.

Add a new user command find for user to filter tasks by inputing keyword matching task description.

Kb-Tay committed:
Update style for main and test files

Kb-Tay committed:
Merge branch 'branch-A-CodingStandard' into branch-A-Javadoc.

Kb-Tay committed:
Merge branch 'branch-Level-9' into branch-A-Javadoc

Kb-Tay committed:
Clean up code and update code style

Kb-Tay committed:
Merge branch 'branch-A-Javadoc'

Kb-Tay committed:
Add a GUI for the chat bot.

Kb-Tay committed:
Merge branch 'branch-Level-10'

Nixx162 commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,55 @@
+     * @return A string description of the event.
+     */
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (from: " + this.from + " to: " + this.to + ")";
+    }
+
+    @Override
+    public String fileString() {
+        return "E" + super.fileString() + " | " + this.from + "-" + this.to;
+    }
```
Using the String.format() method maybe cleaner for these cases

Nixx162 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+
+    /**
+     * Returns the response to the user input. Parses the user input and initialise a new Command,
+     * the Command is then executed to return the correct response.
+     */
```
Might want to add a @return tag here

Nixx162 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,165 @@
+
+    public static final Pattern EVENT_FORMAT = Pattern.compile(
+            "(?<deadline>[^\"]+) /from (" + DATE_PATTERN + ")"
+                    + " /to (" + DATE_PATTERN + ")");
```
There shouldn't be any need for these constants to be public, since they don't seem to be called outside of the Parser class

Nixx162 reviewed (commented):
Some nits, but overall well done. Regarding the directory structure, it might be nice to have command, exception, etc. as sub-packages of the duke package instead. The TaskList class can probably also just be under the task package

AnnabelTing commented on src/test/java/seedu/duke/ParserTest.java:
```diff
@@ -0,0 +1,49 @@
+public class ParserTest {
+
+    @Test
+    public void parse_invalidEditCommandIndex_exceptionThrown() {
```
I like how you named the test!

AnnabelTing commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,181 @@
+
+        String editDesc = "";
+
+        switch (taskType) {
```
Great job on following the coding standard!

AnnabelTing commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,76 @@
+     * Returns string of task to store in file. Description includes
+     * task type, date, if task is done and the task name.
+     */
+    public String fileString() {
```
Since this is a method perhaps you can consider including a verb in its name?

AnnabelTing reviewed (commented):
Overall LGTM

Kb-Tay committed:
Change access of fields in Parser

Kb-Tay committed:
Add assertions and enum for commands

Kb-Tay committed:
Improve readability of Parser

Kb-Tay committed:
Add enums and refactor Parser class

Kb-Tay committed:
Clean up strings returned

Kb-Tay committed:
Add command to remind user of upcoming tasks.

Kb-Tay committed:
Merge pull request #1 from Kb-Tay/branch-A-Assertions

Branch a assertions

Kb-Tay committed:
Merge branch 'master' into branch-A-CodeQuality

Kb-Tay committed:
Merge pull request #2 from Kb-Tay/branch-A-CodeQuality

Branch a code quality

Kb-Tay committed:
Merge branch 'master' into branch-B-Reminders

Kb-Tay committed:
Merge pull request #3 from Kb-Tay/branch-B-Reminders

Branch b reminders

Kb-Tay committed:
Clean up merges in the Parser class.

Kb-Tay committed:
Fix error regarding single word commands.

Previously, commands like bye, list, remind could accept another string behind as an argument.

Commands are now handled to only accept the exact single word command input.

This is to handle invalid commands from user and execute commands only wanted by user.

Kb-Tay committed:
Merge branch 'branch-A-MoreErrorHandling'

Kb-Tay committed:
Update code to pass the Junit testcases.
Currently, some test cases have failed due to changes in code
Fix bugs in the code to ensure test cases pass
Reduce number of bugs in the program

Kb-Tay committed:
Add Junit testing for finding task in TaskList

Kb-Tay committed:
Merge branch 'branch-A-MoreTesting'

Kb-Tay committed:
Error handle invalid dateTime input from user.

Kb-Tay committed:
Edit header comments

Kb-Tay committed:
Add readme and screenshot

Kb-Tay committed:
Fix bugs regarding filepaths and edit code to write to file after each command.

Currently, the filepath was hardcoded and was unable to load from any directory. Writing to file only happened with bye command.

Changed the filepath to be relative to where the program is stored. If file not found, a new directory will be created. Program now writes to file after each command.

This is to ensure user can run jar file no matter where he stores it. Writing to file after each command ensures user progress is saved.

Kb-Tay committed:
Update the readme file.

Kb-Tay committed:
Add hyperlink to readme

Kb-Tay committed:
Changes to error in readme

Kb-Tay committed:
Fix code to work on mac OS
